### ISSUE TITLE
destroyTooltipOnHide={true} is not working in Popover, because the function inside of getPopupContainer is only called on the first click.

### ISSUE THREAD

**Issue by jkdlouis** [ant-design/ant-design#26071]
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/falling-worker-dgl5d?file=/index.js)

### Steps to reproduce
1. Click on the button to open popover.
2. The console.log('fired') only fires on the first click even if destroyTooltipOnHide is set to true.

### What is expected?
If destroyTooltipOnHide is set to true, we expect the function in getPopupContainer to to be called every time we open the popover.

### What is actually happening?
The destroyTooltipOnHide is set to true, and the function in getPopupContainer is only called on the first click to open.

| Environment | Info |
|---|---|
| antd | 4.5.2 |
| React | 16.9.41 |
| System | MacOS Catalina Version 10.15.3 |
| Browser | Safari 13.0.5 |

**Comment by afc163** [comment]
>  getPopupContainer to to be called every time we open the popover.

`destroyTooltipOnHide` don't works in this way.

**Comment by jkdlouis** [comment]
> > getPopupContainer to to be called every time we open the popover.
> 
> `destroyTooltipOnHide` don't works in this way, it simply remove popup dom node after hiding.

But it does not make sense that getPopupContainer's return function doesn't fire on every click to open, is there any work around for this issue?

**Comment by StallionV** [comment]
We too see the same issue happening

**Comment by 07akioni** [comment]
> But it does not make sense that getPopupContainer's return function doesn't fire on every click to open, is there any work around for this issue?

useEffect on `visible`

**Comment by jkdlouis** [comment]
> > But it does not make sense that getPopupContainer's return function doesn't fire on every click to open, is there any work around for this issue?
> 
> `onVisibleChange`?

No, even if I call setState in onVisibleChange, the popover still does not re-render, the getPopupContainer is still only called the first time.

**Comment by afc163** [comment]
I don't understand why does you need to call getPopupContainer multiple times?

**Comment by jkdlouis** [comment]
> I don't understand why does you need to call getPopupContainer multiple times?

Because this popover is used in a table cell, the table height will change base on the table rows, it can be 1 row or 100 rows.  The popover needs to display differently based on the condition, it needs to be attached to the document or parent based on the different condition.

**Comment by StallionV** [comment]
Any updates on this please ?

**Comment by Toril92** [comment]
We have encountered the same problem in our project and the solution, although not a good one, is the most approximate one is to change the key of the component to force react to mount again, and thus call the function getPopupContainer again. In our case, it has been with the Select component.


This is a quick approximation to the solution.

`class OwnSelect extends Component<SelectProps> {

    private attachedElement: HTMLElement | undefined = undefined;

    private keyProp: string | undefined = undefined;

    constructor (props: Readonly<GVSelectProps<ValueType, T>>) {
        super(props);

        this.getPopupContainer = this.getPopupContainer.bind(this);
    }

    existScroll (element: any): boolean {
        if (element.scrollHeight > element.clientHeight) {
            return true;
        } else if (element.parentElement === null) {
            return false;
        }
        return this.existScroll(element.parentElement);
    }

    // eslint-disable-next-line @typescript-eslint/no-explicit-any,@typescript-eslint/explicit-module-boundary-types
    getPopupContainer (trigger: any): HTMLElement {
        if (this.attachedElement === undefined) {
            this.attachedElement = trigger;
        }

        if (this.existScroll(trigger.parentNode)) {
            this.keyProp = trigger.parentNode.tagName;
            return trigger.parentNode;
        }
        this.keyProp = document.body.tagName;
        return document.body;
    }

    render (): ReactNode {
        const { children, errorFeedback, hasError, setFieldValue, onClear, onBlur, ...restProps } = this.props;
        return (
            <Select
				{...restProps}
				key={this.keyProp}
				getPopupContainer={this.getPopupContainer}
				onDropdownVisibleChange={(open): void => {
					if(open && this.attachedElement){
						this.getPopupContainer(this.attachedElement);
					}
				}}
			>
				{children}
			</Select>
        );
    }

}`

**Comment by pabloimrik17** [comment]
any update?

**Comment by afc163** [comment]
Reviewed against the latest `master` (`0b1021909f`). Popup portal containers are now re-evaluated as the component renders.

Latest portal implementation calls getContainer again in its container effect and updates the merged portal target, replacing the one-time v4 behavior.

Closing this issue accordingly. If a minimal reproduction still demonstrates different behavior on the current `master`, please open a new issue with that reproduction.
